dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/BrowserAPI.java

```diff
@@ -254,256 +248,111 @@ public WfData(Contentlet contentlet, List<Integer> permissions, User user, boole
+	                linkMap.put("hasTitleImage", "");
+	                linkMap.put("__icon__", "linkIcon");
+	                returnList.add(linkMap);
+	            }
+	        }
 
-			if (parent == null) {
+            // Getting the html pages directly under the parent folder or host
+            List<IHTMLPage> pages = new ArrayList<IHTMLPage>();
+            
+            if (parent != null) {//For folders
+                if(!showWorking) {
+                    pages.addAll(APILocator.getHTMLPageAssetAPI().getLiveHTMLPages(parent, user, false));
+                }
+                else {
+                    pages.addAll(APILocator.getHTMLPageAssetAPI().getWorkingHTMLPages(parent, user, false));
+                }
 
-				// Getting the files directly under the host
-				files.addAll(APILocator.getFileAssetAPI().findFileAssetsByHost(host, user, !showWorking, showWorking, false, false));
-				if(showArchived)
-					files.addAll(APILocator.getFileAssetAPI().findFileAssetsByHost(host, user, !showWorking, showWorking, showArchived, false));
+                if(showArchived) {
+                    pages.addAll(APILocator.getHTMLPageAssetAPI().getDeletedHTMLPages(parent, user, false));
+                }
+            } else {//For hosts
+                if ( !showWorking ) {
+                    pages.addAll( APILocator.getHTMLPageAssetAPI().getLiveHTMLPages( host, user, false ) );
+                } else {
+                    pages.addAll( APILocator.getHTMLPageAssetAPI().getWorkingHTMLPages( host, user, false ) );
+                }
 
-			} else {
-		        ChildrenCondition cond = new ChildrenCondition();
-		        
-		        if(showWorking)
-		        	cond.working = true;
-		        else
-		        	cond.live = true;
-				
-		        cond.deleted = showArchived;
+                if ( showArchived ) {
+                    pages.addAll( APILocator.getHTMLPageAssetAPI().getDeletedHTMLPages( host, user, false ) );
+                }
+            }
 
-				files.addAll(APILocator.getFileAssetAPI().findFileAssetsByFolder(parent, "", !showWorking, showWorking, user, false));
-			}
+            for (IHTMLPage page : pages) {
+                // include only pages for the given language
+                if(languageId > 0 && ((Contentlet)page).getLanguageId()!= languageId) {
+                    continue;
+                }
+                returnList.add(htmlPageMap((HTMLPageAsset) page, user, showArchived, languageId));
+            }
+		}
+		
 
-			//remove duplicated legacy files from list. See issue 5943
-    		HashSet<Versionable> tempFilesListToSet = new HashSet<Versionable>(files);
-    		files.clear();
-    		files.addAll(tempFilesListToSet);
 
-		} catch (Exception e2) {
-			Logger.error(this, "Could not load files : ", e2);
+		
+		String luceneQuery=(dotAssets)  ? "+basetype:(4 OR 9) " : "+basetype:4 ";
+		luceneQuery+=(languageId>0) ? " +languageid:" +  languageId : "";
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Prefer StringBuilder (non-synchronized) or StringBuffer (synchronized) over += for concatenating strings](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5337824)